### PR TITLE
Hard: automatic liquidation by LTV index

### PR DESCRIPTION
**Pull request by denalimarsh**
This PR builds on the LTV index scaffolding, introducing the actual liquidation engine that's run by the BeginBlocker. Currently, this PR only contains logic.

TODO:
- [ ] feature testing

The base branch is set to `dm-hard-index-liquidations` because the -> `master` commit history is messy. Will change to master for the merge.

### CODE REVIEW

File: x/harvest/keeper/liquidation.go [hunk 1]
```diff
@@ -16,23 +18,29 @@ type LiqData struct {
...
-	//			(?) Removing borrow from the store
-	//			(?) Removing borrow LTV from LTV index
+	params := k.GetParams(ctx)
+	borrowers := k.GetLtvIndexSlice(ctx, params.CheckLtvIndexCount)
+
+	// Sort items in slice to ensure deterministic auction order
+	sortedBorrowers := addressSort(borrowers)
 
+	for _, borrower := range sortedBorrowers {
+		k.AttemptKeeperLiquidation(ctx, sdk.AccAddress(types.LiquidatorAccount), borrower)
```
Comment (karzak) [inline x/harvest/keeper/liquidation.go:28]: I would check the error value here and panic if an error is found other than ErrBorrowNotLiquidatable

File: x/harvest/keeper/liquidation.go [hunk 2]
```diff
@@ -16,23 +18,29 @@ type LiqData struct {
...
-	//		Use current asset prices from pricefeed to calculate current LTV for each asset
-	//		If LTV of any asset is over the max, liquidate it by
-	//			Sending coins to auction module
-	//			(?) Removing borrow from the store
-	//			(?) Removing borrow LTV from LTV index
+	params := k.GetParams(ctx)
+	borrowers := k.GetLtvIndexSlice(ctx, params.CheckLtvIndexCount)
+
+	// Sort items in slice to ensure deterministic auction order
+	sortedBorrowers := addressSort(borrowers)
```
Comment (karzak) [inline x/harvest/keeper/liquidation.go:25]: I don't think sorting is necessary for deterministic order

**Review by karzak — CHANGES_REQUESTED** [review]
Looking good, just a few minor observations

File: x/harvest/keeper/liquidation.go (hunk 1 above)
Comment (denalimarsh) [reply x/harvest/keeper/liquidation.go]: Added in https://github.com/Kava-Labs/kava/pull/743/commits/7f90920acafd1cdc5178df19a9a07d08bbda6a66

File: x/harvest/keeper/liquidation.go (hunk 2 above)
Comment (denalimarsh) [reply x/harvest/keeper/liquidation.go]: Removed in https://github.com/Kava-Labs/kava/pull/743/commits/7f90920acafd1cdc5178df19a9a07d08bbda6a66, https://github.com/Kava-Labs/kava/pull/743/commits/d110c36bbeb34abd432abae77e8b446efbe29f85

**Review by karzak — APPROVED** [review]

**Pull request merged**